chat.go

```diff
@@ -43,12 +51,68 @@ type ChatCompletionRequest struct {
 	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
+	Functions        []*FunctionsDefine      `json:"functions,omitempty"`
+	FunctionCall     string                  `json:"function_call,omitempty"`
 }
 
+type FunctionsDefine struct {
+	Name        string      `json:"name,omitempty"`
+	Description string      `json:"description,omitempty"`
+	Parameters  *FuncParams `json:"parameters,omitempty"`
```

Comment:
@Ccheers The official documentation says `parameters` is optional, but when you actually call the API, it says `parameters` is required.

```
$ curl https://api.openai.com/v1/chat/completions -u :$OPENAI_API_KEY -H 'Content-Type: application/json' -d '{
  "model": "gpt-3.5-turbo-0613",
  "messages": [
    {"role": "user", "content": "What is the weather like in Boston?"}
  ],
  "functions": [
    {
      "name": "get_current_weather"
    }
  ]
}'
```

Response:
```json
{
   "error":{
      "message":"'parameters' is a required property - 'functions.0'",
      "type":"invalid_request_error",
      "param":null,
      "code":null
   }
}
```

Reply:
You are right and the type must be "object", but as a api we shouldn't limit it. Function `CreateChatCompletionWithFunctionCall` will do it

Reply:
we can add some comment to user